transform.py

```diff
@@ -0,0 +1,169 @@
+"""Python script to transform the data extracted from the National Archives 
+website through ChatGPT into a format that can be loaded into a database"""
+
+from os import getenv
+from datetime import datetime, date
+from string import capwords
+from ast import literal_eval
+from dotenv import load_dotenv
+from openai import OpenAI
+import tiktoken
+import prompts
+from extract import get_listing_data
+
+
+load_dotenv()
+client = OpenAI(api_key=getenv("OPENAI_API_KEY"))
+
+
+def shorten_text_by_tokens(
+    text, keep_start_tokens=4000, keep_end_tokens=4000, placeholder="[...]"
+):
+    """Shortens court transcript by removing tokens from the middle"""
+    encoder = tiktoken.encoding_for_model("gpt-4o-mini")
+    tokens = encoder.encode(text)
+
+    if len(tokens) <= keep_start_tokens + keep_end_tokens:
+        return text
+
+    start_tokens = tokens[:keep_start_tokens]
+    end_tokens = tokens[-keep_end_tokens:]
+
+    start_text = encoder.decode(start_tokens)
+    end_text = encoder.decode(end_tokens)
+
+    shortened_text = f"{start_text}{placeholder}{end_text}"
+
+    return shortened_text
+
+
+def get_summary(prompt: str, transcript: str):
+    """Collect data about the transcript using the GPT-4o-mini model"""
+    completion = client.chat.completions.create(
+        model="gpt-4o-mini",
+        messages=[
+            {"role": "system", "content": prompt},
+            {"role": "user", "content": transcript},
+        ],
+        temperature=0.1,
```

Comment:
A small comment explaining temperature, this is actually my fault.

Reply:
Will add this, thanks Hamoodi.